src/Lib.hs

```diff
@@ -20,17 +27,13 @@ startPetSelection user goldRemaining = do
     pet2 <- getPet allPets
     pet3 <- getPet allPets
     let petChoices = [pet1, pet2, pet3]
-    putStrLn $ "1. " ++ show (petChoices !! 0)
-    putStrLn $ "2. " ++ show (petChoices !! 1)
-    putStrLn $ "3. " ++ show (petChoices !! 2)
+    mapM_ (\(idx, choice) -> putStrLn $ show idx ++ ". " ++ display choice) (zip [1..] petChoices) 
```

Comment:
Might be worth extracting this to a well named function. That way we capture intent and isolate how "funky" this looks a smaller code block.